addons/web/static/tests/views/sample_server_tests.js

```diff
@@ -323,16 +352,19 @@ odoo.define('web.sample_server_tests', function (require) {
                 fields: [],
                 groupBy: ['profession'],
             });
+            const groups = fillRandomGroups(result, "profession",
```

Comment:
I think this is too artificial. We should accept the fact that it may produce empty groups, which are thus not returned. I think the asserts should be adapted to tolerate this.
What we know:
   (1) there will be at least one group
   (2) the total number of records in all groups is MAIN_RECORDSET_SIZE
   (3) groups are either 'gardener', 'brewer' or 'adventurer'

How can we assert that:
   (1) assert.ok(groups.length >= 1);
   (2) like already done
   (3) assert.ok(groups.every(g => ["gardener", ...].includes(g.profession)));

I agree that this is a bit disturbing that we can't do fully deterministic asserts like we always do (e.g. x === n), and I'm not happy about it, but this is part of the SampleServer's nature. However, I believe that it is important to test what we **know**, even if it leads to unusual asserts, instead of tweaking the results, which might hide issues. For instance with your solution, read_group could return nothing and the test would pass.

Reply:
Maybe a more elegant alternative would be to "remove" the random aspects in the tests. Good thing is, the SampleServer has a `_getRandomSubRecordId` function that we could override, to generate whatever we want. See this [interesting blog](https://jessitron.com/2013/08/05/a-trick-for-deterministic-testing-of-random-behavior/)

Reply:
Done